client/app/components/sections/SearchPage/SearchPage.js

```diff
@@ -24,8 +25,11 @@ class SearchPage extends Component {
   }
 
   listingProps(listing, color) {
+
+    // DiscoveryAPI doesn't return UUID yet but a string
+    const listingKey = listing.id instanceof UUID ? listing.id.value : listing.id;
```

Comment:
String is a perfectly valid UUID representation. I think this should be handled in `transitImmutableConverter`, and have a shared way of turning UUIDs to strings.